core/modules/qdisp/Executive.cc

```diff
@@ -475,41 +366,43 @@ void Executive::_unTrack(int jobId) {
     }
 }
 
-/// This function only acts when there are errors. In there are no errors,
-/// markCompleted() does the cleanup, while we are waiting (in
-/// _waitAllUntilEmpty()).
+/// Remove all jobs from the _incompleteJobs map that have errors.
+// This function only acts when there are errors. In there are no errors,
+// markCompleted() does the cleanup, while we are waiting (in _waitAllUntilEmpty()).
 void Executive::_reapRequesters(std::unique_lock<std::mutex> const&) {
-    RequesterMap::iterator i, e;
-    while(true) {
-        bool reaped = false;
-        for(i=_requesters.begin(), e=_requesters.end(); i != e; ++i) {
-            if(!i->second->getError().isNone()) {
-                // Requester should have logged the error to the messageStore
-                LOGF_INFO("Executive (%1%) reaped requester for jobId=%2%"
-                          % (void*)this % i->first);
-                _requesters.erase(i);
-                reaped = true;
-                break;
-            }
-        }
-        if(!reaped) {
-            break;
+    for(auto i=_incompleteJobs.begin(), e=_incompleteJobs.end(); i != e;) {
+        if(!i->second->getDescription().respHandler()->getError().isNone()) {
+            // Requester should have logged the error to the messageStore
+            LOGF_INFO("Executive (%1%) reaped requester for jobId=%2%" % (void*)this % i->first);
+            auto eraseMe = i;
+            ++i;
+            _incompleteJobs.erase(eraseMe);
+        } else {
+            ++i;
```

Comment:
I think it has to stay in the else case. i = _incompleteJobs.erase(i) should increment the iterator to the next object. another increment will skip things.